components/main/Desktop/ExportButton.jsx

```diff
@@ -48,22 +49,60 @@ function ExportButton({ filters }) {
       .map(v => `'${v.typeName}'`)
       .join(', ');
 
-    // in the case user chooses one neighborhood or all are selected + dates and status
-    const neighborhoodDataQuery = `SELECT * FROM requests_2024 
-      WHERE CreatedDate >= '${filters.startDate}' 
-      AND CreatedDate < '${filters.endDate}'
-      ${requestStatusFilter !== '' ? ` AND Status='${requestStatusFilter}'` : ''}
-      ${filters.councilId !== null ? ` AND NC='${filters.councilId}'` : ''} 
-      AND RequestType IN (${formattedRequestTypes});`;
-
-    const groupedAddressQuery = `SELECT Address, COUNT(*) AS NumberOfRequests FROM requests_2024
-      WHERE CreatedDate >= '${filters.startDate}' 
-      AND CreatedDate < '${filters.endDate}' 
-      AND Status = 'Open' 
-      AND NC = '${filters.councilId}' 
-      AND RequestType IN (${formattedRequestTypes})
-      GROUP BY Address`;
+    const startYear = moment(filters.startDate).year();
+    const endYear = moment(filters.endDate).year();
+
+    const generateQuery = (grouped = false) => {
+      if (startYear === endYear) {
+        if (grouped) {
+          return `SELECT Address, COUNT(*) AS NumberOfRequests FROM requests_${startYear}
+            WHERE CreatedDate >= '${filters.startDate}' 
+            AND CreatedDate < '${filters.endDate}' 
+            AND Status = 'Open' 
+            AND NC = '${filters.councilId}' 
+            AND RequestType IN (${formattedRequestTypes})
+            GROUP BY Address`;
+        }
+        return `SELECT * FROM requests_${startYear} 
```

Comment:
I would either convert this logic block to explicitly be if-else, so readers  can see `startYear === endYear` and `grouped` are the deciding factors between which logical pathway is taken. OR I would add comments indicating what the state of logic should be...
i.e. here on line 66: `//here, our data comes from the same year and we wish to obtain all rows that match the filters

Reply:
While we're considering comments, here's my other request for adding a note: https://github.com/hackforla/311-data/issues/1746#issuecomment-2159607947